### ISSUE TITLE
ERROR: Request failed! Error code ENETUNREACH.

### ISSUE THREAD

**Issue by RPJacobs** [jmichas/homebridge-zway-plugin#9]
After a couple of hours I get this error message.... devices are unavailable.

ERROR: Request failed! Error code ENETUNREACH. Check the URL in config.json and ensure that the URL can be reached from this system!

<img width="1048" alt="Screenshot 2024-01-26 at 21 22 15" src="https://github.com/jmichas/homebridge-zway-plugin/assets/14312145/ee8c59a1-f6a7-4906-9a0d-7ed2a329a5d5">

After a restart all is fine.

<img width="954" alt="Screenshot 2024-01-26 at 21 23 32" src="https://github.com/jmichas/homebridge-zway-plugin/assets/14312145/2c85806b-6982-42ec-a59e-e169b3480323">

**Comment by jmichas** [comment]
Hi, are you using my version of the plugin? This was an error that happened in the old code that I had forked from, it was fixed.

**Comment by RPJacobs** [comment]
Yes I use your plugin. Equally I used the other plugin before and had a username password error. You fixed that one with the cookie. 

![image](https://github.com/jmichas/homebridge-zway-plugin/assets/14312145/c32bc735-f7df-459a-9e89-4c601c5d4b3d)

**Comment by jmichas** [comment]
I will look through the code to see if I can figure out what that error code really means. Sorry, it is sometimes tough because I am not intimately familiar with existing code.
Is there a time frame this starts to occur? Like is it hours after a restart, days, weeks? Or just random?

**Comment by jmichas** [comment]
Actually, also, what version of homebridge and node are you running?
This has only been tested on HB 1.7 and node 18.16, so if youa re running a more recent version that may be the issue.

**Comment by RPJacobs** [comment]
![image](https://github.com/jmichas/homebridge-zway-plugin/assets/14312145/c60f8d26-a35a-43f8-af97-065d76560b7b)

It's a fresh install pi4B. 

![image](https://github.com/jmichas/homebridge-zway-plugin/assets/14312145/2ead968c-321b-48da-a8f4-3848021bc423)

**Comment by jmichas** [comment]
Yeah, that's what I thought. I don't currently have an environment to test against the latest node/HB.
If a restart fixes the problem a workaround for now could be to just set your device to reboot in the middle of the night using cron. It is pretty simple to setup, you can google how to use cron. I used to have my pi reboot at 3am every night to fix the old problem before we fixed it properly in this repo. As soon as I can I will look into setting up some sort of VM using latest node to code and test against.

**Comment by RPJacobs** [comment]
restart did not work for me.  I downgraded to Node.js 18 an no more errors.

**Comment by RPJacobs** [comment]
Moved to a docker image with node.js 18 to test:

![Screenshot 2024-02-01 at 14 10 28](https://github.com/jmichas/homebridge-zway-plugin/assets/14312145/0599e7f1-0d69-46b1-ac3d-3e8c14c39990)

The (almost the same) error is still there but only for a minute and then it recovers..

![Screenshot 2024-02-01 at 14 10 56](https://github.com/jmichas/homebridge-zway-plugin/assets/14312145/7a6f1830-df93-4d3f-bf83-8a7ab9a0a768)

![Screenshot 2024-02-01 at 14 11 08](https://github.com/jmichas/homebridge-zway-plugin/assets/14312145/ad7ec444-ec70-4035-a3d5-8839d8e57095)

**Comment by RPJacobs** [comment]
don't think it's a session error locking at the session log in z-way-server. The last session was created a day before the error....

![Screenshot 2024-02-01 at 14 18 57](https://github.com/jmichas/homebridge-zway-plugin/assets/14312145/eeafbcdb-6eee-4c94-9e16-45b182792501)
